### PR TITLE
net: openthread: Fix possible deadlock in net_mgmt handlers

### PR DESCRIPTION
**Pull request by rlubos**
The core part of this PR is fixing a deadlock situation in OpenThread's net_mgmt handler, please see the commit message for explaination.

Additionally, Small improvement that make sense IMO are proposed (see the other commit). Altogether those fixes allow to use mDNS responder with OT.

### CODE REVIEW

**Comment by rlubos** [comment]
CC @lmaciejonczyk @canisLupus1313

**Review by canisLupus1313 — DISMISSED** [review]

**Comment by rlubos** [comment]
Dropped the select on `NET_MGMT`, apparently for simple use cases (RCP) NET MGMT is not a must.

File: subsys/net/l2/openthread/openthread_utils.c [hunk 1]
```diff
@@ -157,42 +157,52 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
...
+	 * address added event.
+	 */
+	ipv6 = context->iface->config.ip.ipv6;
+	if (ipv6 == NULL) {
+		NET_ERR("No IPv6 container allocated");
 		return;
 	}
 
-	/* save the last added IP address for this interface */
+	/* Find the net_if_addr strcture containing the newly added address. */
```
Comment (pdgendt) [inline subsys/net/l2/openthread/openthread_utils.c:177]: Nit: typo

File: subsys/net/l2/openthread/openthread_utils.c [hunk 2]
```diff
@@ -157,42 +157,52 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 	}
 }
 
-void add_ipv6_addr_to_ot(struct openthread_context *context)
+void add_ipv6_addr_to_ot(struct openthread_context *context,
+			 const struct in6_addr* addr6)
```
Comment (pdgendt) [inline subsys/net/l2/openthread/openthread_utils.c:161]: `addr6` isn't checked for `NULL` inside this function, which feels dangerous as it is exposed in the header.

Comment (rlubos) [reply subsys/net/l2/openthread/openthread_utils.c]: It's a private header though ,it's not exposed to the application. The pointer is verified before function call, I'm not sure if we should repeat here (we don't verify `context` either).

**Review by pdgendt — DISMISSED** [review]

**Review by lmaciejonczyk — DISMISSED** [review]

**Review by edmont — DISMISSED** [review]

File: subsys/net/l2/openthread/openthread.c [hunk 3]
```diff
@@ -122,10 +122,14 @@ static void ipv6_addr_event_handler(struct net_mgmt_event_callback *cb,
 		return;
```
Comment (jukkar) [inline subsys/net/l2/openthread/openthread.c:122]: Two typos in commit msg `s/withing/within'

**Review by jukkar — COMMENTED** [review]
LGTM, small nit in commit msg

**Review by canisLupus1313 — DISMISSED** [review]

File: subsys/net/l2/openthread/openthread.c (hunk 3 above)
Comment (rlubos) [reply subsys/net/l2/openthread/openthread.c:122]: Thanks, fixed

**Review by jukkar — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**